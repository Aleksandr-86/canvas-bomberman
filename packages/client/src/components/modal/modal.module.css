.modal {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: rgb(0 0 0 / 20%);
  backdrop-filter: blur(2px);
}

.modalActive {
  display: flex;
  justify-content: center;
  align-items: center;
}

.modalClose {
  display: none;
}

.modalContent {
  z-index: 999999;
  min-width: 300px;
}
