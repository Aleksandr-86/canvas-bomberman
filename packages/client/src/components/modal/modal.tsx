import React from 'react'
import styles from './modal.module.css'

interface Props {
  isActive?: boolean
  setActive?: any
  children?: any
}

export const Modal: React.FC<Props> = props => {
  const { children, isActive, setActive } = props

  const handleChange = () => {
    setActive(false)
  }

  return (
    <div
      className={
        isActive
          ? `${styles.modal} ${styles.modalActive}`
          : `${styles.modal} ${styles.modalClose}`
      }
      onClick={handleChange}>
      <div className={styles.modalContent} onClick={e => e.stopPropagation()}>
        {children}
      </div>
    </div>
  )
}
