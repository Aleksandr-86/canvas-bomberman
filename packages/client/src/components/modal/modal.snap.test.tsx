import * as renderer from 'react-test-renderer'
import { Modal } from './modal'

describe(`<Modal/> snapshot test cases`, () => {
  test(`<Modal/> should render <div className="modal modalActive"> parent with child elements (default)`, () => {
    const tree = renderer.create(<Modal />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test(`<Modal/> should render <div className="modal modalClose"> parent with child elements (isActive false)`, () => {
    const tree = renderer.create(<Modal isActive={false} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test(`<Modal/> should render <div className="modal modalActive"> parent with childrenElement`, () => {
    const tree = renderer
      .create(<Modal>{<div>childrenElement</div>}</Modal>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
