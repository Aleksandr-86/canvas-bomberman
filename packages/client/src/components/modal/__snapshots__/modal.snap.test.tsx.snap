// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal/> snapshot test cases <Modal/> should render <div className="modal modalActive"> parent with child elements (default) 1`] = `
<div
  className="modal modalClose"
  onClick={[Function]}
>
  <div
    className="modalContent"
    onClick={[Function]}
  />
</div>
`;

exports[`<Modal/> snapshot test cases <Modal/> should render <div className="modal modalActive"> parent with childrenElement 1`] = `
<div
  className="modal modalClose"
  onClick={[Function]}
>
  <div
    className="modalContent"
    onClick={[Function]}
  >
    <div>
      childrenElement
    </div>
  </div>
</div>
`;

exports[`<Modal/> snapshot test cases <Modal/> should render <div className="modal modalClose"> parent with child elements (isActive false) 1`] = `
<div
  className="modal modalClose"
  onClick={[Function]}
>
  <div
    className="modalContent"
    onClick={[Function]}
  />
</div>
`;
